go-controller/pkg/util/multi_network_test.go

```diff
@@ -603,6 +606,102 @@ func TestIsDefault(t *testing.T) {
 	}
 }
 
+func TestGetPodNADToNetworkMapping(t *testing.T) {
+	const (
+		attachmentName = "attachment1"
+		namespaceName  = "ns1"
+		networkName    = "l3-network"
+	)
+
+	type testConfig struct {
+		desc                          string
+		inputNamespace                string
+		inputNetConf                  *ovncnitypes.NetConf
+		inputPodAnnotations           map[string]string
+		expectedError                 error
+		expectedIsAttachmentRequested bool
+	}
+
+	tests := []testConfig{
+		{
+			desc:                "Looking for a network *not* present in the pod's attachment requests",
+			inputNamespace:      namespaceName,
+			inputPodAnnotations: map[string]string{nadv1.NetworkAttachmentAnnot: "[]"},
+			inputNetConf: &ovncnitypes.NetConf{
+				NetConf:  cnitypes.NetConf{Name: networkName},
+				Topology: ovntypes.Layer3Topology,
+				NADName:  GetNADName(namespaceName, attachmentName),
+			},
+			expectedIsAttachmentRequested: false,
+		},
+		{
+			desc: "Looking for a network present in the pod's attachment requests",
+			inputNetConf: &ovncnitypes.NetConf{
+				NetConf:  cnitypes.NetConf{Name: networkName},
+				Topology: ovntypes.Layer3Topology,
+				NADName:  GetNADName(namespaceName, attachmentName),
+			},
+			inputPodAnnotations: map[string]string{
+				nadv1.NetworkAttachmentAnnot: GetNADName(namespaceName, attachmentName),
+			},
+			expectedIsAttachmentRequested: true,
+		},
+		{
+			desc:           "Multiple attachments to the same network in the same pod are not supported",
```

Comment:
Where are the multiple attach? I'm on my phone so might be missing something here

Reply:
> Where are the multiple attach? I'm on my phone so might be missing something here

It's below, on line 658 - it would render to something like:
`"k8s.v1.cni.cncf.io/networks": "<namespaceName>/<attachmentName>,namespaceName>/<attachmentName>"`

```golang
inputPodAnnotations: map[string]string{
				nadv1.NetworkAttachmentAnnot: fmt.Sprintf("%[1]s,%[1]s", GetNADName(namespaceName, attachmentName)),
			},
```

Reply:
AHHHHH dangit I missed that %[1]s,%[1]s trick lol ack gotcha